inc/logger/LogStream.h

```diff
@@ -0,0 +1,92 @@
+//MIT License
+
+//Copyright (c) 2020 bexoft GmbH (mail@bexoft.de)
+
+//Permission is hereby granted, free of charge, to any person obtaining a copy
+//of this software and associated documentation files (the "Software"), to deal
+//in the Software without restriction, including without limitation the rights
+//to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//copies of the Software, and to permit persons to whom the Software is
+//furnished to do so, subject to the following conditions:
+
+//The above copyright notice and this permission notice shall be included in all
+//copies or substantial portions of the Software.
+
+//THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//SOFTWARE.
+
+#pragma once
+
+#include "Logger.h"
+#include "helpers/BexDefines.h"
+#include <string>
+#include <sstream>
+
+
+
+#define STREAM_BASIC(level)     static const finalmq::LogContext TOKENPASTE2(context, __LINE__){level, MODULENAME, METHODNAME, __FILE__, __LINE__}; finalmq::LogStream::getStream(TOKENPASTE2(context, __LINE__))
```

Comment:
Would it produce a global state?

Reply:
static const LogContext context__LINE__
is a local variable where ever a log entry is defined. To have unique variable names the line number is part of the variable name.